feat: only require screen capture and accessibility permissions

r17x opened the pull request:
## Changes

* Removed Audio (Microphone) Request permission on first-run.
* Request permission when user click Audio (Microphone) Selection - (only show when not `granted`).

### Attachments

| Granted | Not Granted |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/9bc1e6fd-a35b-448d-a416-09867eb6838c) | ![image](https://github.com/user-attachments/assets/ed768474-e723-4987-a4a2-d5dcd0c86e1b) |
|  ![image](https://github.com/user-attachments/assets/2700052e-64c4-4bb6-ae2f-cbe1779014e1) | |

/claim #91

Ready for review @richiemcilroy

richiemcilroy commented:
Hey @r17x this is awesome! Thanks for submitting so quickly. 

Would you be able to also do this for webcam as well?

Make it so that camera permission is not required at startup, until you decide to (by clicking the list). 

Also, willing to add another $100 to the bounty if you can get the green MacBook webcam light to stop showing when no webcam is currently selected.

r17x commented:
Already updated the PR @richiemcilroy 

Here's videos for another $100 bounty.

https://github.com/user-attachments/assets/28500477-9a33-43ea-bdc3-8ca8883d7d74

r17x commented:
Heres for camera (same behavior with `microphone`). @richiemcilroy 

https://github.com/user-attachments/assets/1dfdf7b8-599b-4915-bc5e-7aa11f0e041e

richiemcilroy commented:
Hey @r17x! looks awesome :)

One problem I've noticed is that once you do give permission, you only see the devices (webcam etc) once you right click and refresh.

Is there a way to make it so that we check again/refresh the list once the permission has been granted?

Also if it's possible to not open the dropdown list when we're actually requesting the permission.

So technically the entire "No Camera" button should request for permission on click, instead of just "Request Permission".

**Actions:**
	1.	Fix Device List Update: Automatically refresh the list of devices (webcam, etc.) once permission is granted, without requiring a manual right-click or refresh.
	2.	Prevent Dropdown from Opening During Permission Request: Ensure that the dropdown list for devices doesn’t open while the app is requesting camera permissions.
	3.	Update “No Camera” Button Behavior: Modify the “No Camera” button to request permission when clicked, instead of only the “Request Permission” button initiating the permission request.

galabra commented on apps/desktop/src/routes/(window-chrome)/index.tsx:
```diff
@@ -292,12 +301,21 @@ export default function () {
+              if (device) {
+                commands.setRecordingOptions({
+                  ...options.data,
+                  cameraLabel: device && device?.deviceId ? device?.label : null,
```
`device && device?.deviceId` is equivalent to `device?.deviceId`

r17x replied:
Nice catch, thank you @galabra

richiemcilroy commented:
Let me know your thoughts @r17x whenever you get a chance - would love to get this live ASAP

r17x commented:
@richiemcilroy Here's my new updates:

> 1. Fix Device List Update: Automatically refresh the list of devices (webcam, etc.) once permission is granted, without requiring a manual right-click or refresh.

Automatically update the list options devices (you must be click the options lists for showing the list of devices). Anyway, I didn't understood what's you mean `right-click` or `refresh`. 

> 2. Prevent Dropdown from Opening During Permission Request: Ensure that the dropdown list for devices doesn’t open while the app is requesting camera permissions.

Fixed

> 3. Update “No Camera” Button Behavior: Modify the “No Camera” button to request permission when clicked, instead of only the “Request Permission” button initiating the permission request.

Fixed


https://github.com/user-attachments/assets/fbac08c3-638f-4a08-ba9c-8e57eb7e9276

r17x commented on apps/desktop/src/utils/queries.ts:
```diff
@@ -51,13 +61,42 @@ export function createWindowsQuery() {
 
 export function createAudioDevicesQuery() {
   const devices = createQuery(() => listAudioDevices);
-  createTimer(() => devices.refetch(), 1000, setInterval);
```
I removed this and set using `queryOptions -> refetchInterval: 1000` - same way
